src/test/suite/extension.test.ts

```diff
@@ -31,6 +34,74 @@ suite('command', () => {
     })
 })
 
+suite('manifestMap', () => {
+    function createTestManifest(uri: Uri): Manifest {
+        return new Manifest(uri, {
+            id: path.parse(uri.fsPath).name,
+            modules: [],
+            sdk: '',
+            runtime: '',
+            'runtime-version': '',
+            command: '',
+            'finish-args': [],
+        })
+    }
+
+    test('add, delete, get, getFirstItem, isEmpty, & size', function () {
+        const map = new ManifestMap()
+        assert(map.isEmpty())
+        assert.equal(map.getFirstItem(), undefined)
+
+        const manifestAUri = Uri.file('/home/test/a.a.a.json')
+        const manifestA = createTestManifest(manifestAUri)
+        map.add(manifestA)
+        assert.equal(map.size(), 1)
+        assert.deepStrictEqual(map.get(manifestAUri), manifestA)
+        assert.deepStrictEqual(map.getFirstItem(), manifestA)
+
+        const manifestBUri = Uri.file('/home/test/b.b.b.json')
+        const manifestB = createTestManifest(manifestBUri)
+        map.add(manifestB)
+        assert.equal(map.size(), 2)
+        assert.deepStrictEqual(map.get(manifestBUri), manifestB)
+        assert.deepStrictEqual(map.getFirstItem(), manifestA)
+
+        assert(map.delete(manifestAUri))
+        assert.equal(map.size(), 1)
+        assert.equal(map.get(manifestAUri), undefined)
+
+        assert.deepStrictEqual(map.getFirstItem(), manifestB)
+
+        assert(map.delete(manifestBUri))
+        assert.equal(map.size(), 0)
+        assert.equal(map.get(manifestBUri), undefined)
+
+        assert(map.isEmpty())
+        assert.equal(map.getFirstItem(), undefined)
+    })
+
+    test('forEach', function () {
+        const map = new ManifestMap()
+
+        map.forEach(() => assert(false))
```

Comment:
What is the point of this assertion?

Reply:
that forEach would not never call the callback

Reply:
there's probably more explicit way to fail a test